src/backend/joanie/payment/factories.py

```diff
@@ -0,0 +1,138 @@
+"""
+Payment application factories
+"""
+import random
+import string
+from decimal import Decimal as D
+
+import factory
+import factory.fuzzy
+import payplug
+
+from joanie.core.factories import OrderFactory, UserFactory
+
+from . import models
+
+
+class CreditCardFactory(factory.django.DjangoModelFactory):
+    """A factory to create a credit card."""
+
+    class Meta:
+        """Meta"""
+
+        model = models.CreditCard
+
+    brand = factory.Faker("credit_card_provider")
+    expiration_month = factory.Faker("credit_card_expire", date_format="%m")
+    expiration_year = factory.Faker("credit_card_expire", date_format="%Y")
+    last_numbers = factory.fuzzy.FuzzyText(length=4, chars=string.digits)
+    owner = factory.SubFactory(UserFactory)
+    title = factory.Faker("name")
+    token = factory.Sequence(lambda k: f"card_{k:022d}")
+
+
+class InvoiceFactory(factory.django.DjangoModelFactory):
+    """A factory to create an invoice"""
+
+    class Meta:
+        """Meta"""
+
+        model = models.Invoice
+
+    amount = factory.LazyAttribute(lambda i: i.order.product.price)
+    order = factory.SubFactory(OrderFactory)
+    recipient_address = factory.Faker("address")
+    recipient_name = factory.Faker("name")
+
+
+class TransactionFactory(factory.django.DjangoModelFactory):
+    """A factory to create a transaction"""
+
+    class Meta:
+        """Meta"""
+
+        model = models.Transaction
+
+    reference = factory.LazyAttributeSequence(
+        lambda t, n: f"{'ref' if t.is_credit else 'pay'}_{n:05d}"
+    )
+    invoice = None
+    amount = factory.LazyAttribute(
+        lambda t: D(random.randrange(t.invoice.amount * 100)) / 100  # nosec
+    )
+    is_credit = factory.fuzzy.FuzzyChoice([True, False])
+
+
+class PayplugBillingFactory(factory.Factory):
```

Comment:
I'm questioning if these models should include "Payplug" in their name or if they would also work with another backend...

Reply:
There are always small differences and I'm pretty sure that if we need to implement a new payment backend we will not be able to reuse these factories.

Reply:
But the model is also specific ?

```
model = payplug.resources.Payment.Billing
```

Reply:
think it's highly preferable if the model (and thus factories) are determined in joanie and the adaptation to the backend data format is done in backend code.

Reply:
Not sure to take your point. `Payplug*` factories has been created only to mock responses incoming from payplug server (when a notification is received and when a payment is created). In fact, when we received this notification, payplug backend is in charge to parse this date and adapt it to our model.

Maybe I should move factories related to Payplug in a dedicated module?

Reply:
+1 for moving it to the backends/payplug directory.